locations/spiders/stadt_zuerich_ch.py

```diff
@@ -0,0 +1,201 @@
+            return {
+                "addr:street": street,
+                "addr:housenumber": housenumber,
+                "addr:postcode": p.get("plz"),
+                "nohousenumber": "yes" if not housenumber else None,
+            }
+        if addr := p.get("adresse"):
+            return {"addr:full": f"{addr}, Zürich"}
+        return {}
+
+    def parse_bicycle_parking(self, p):
+        amenity = {
+            "Motorrad": "motorcycle_parking",
+            "Velo": "bicycle_parking",
+            "Beide": "bicycle_parking;motorcycle_parking",
+        }.get(p["name"])
+        operator, operator_wikidata = self.operators["Tiefbauamt"]
+        return {
+            "amenity": amenity,
+            "capacity": int(p.get("anzahl_pp", 0)) or None,
+            "operator": operator,
+            "operator:wikidata": operator_wikidata,
+        }
+
+    def parse_name(self, p):
+        names = [p.get(n) for n in ("name", "namenzus")]
+        return ", ".join([n for n in names if n])
+
+    def parse_note(self, p):
+        lines = (p.get("bemerkung") or "").split("<BR>")
+        lines.extend((p.get("kommentar") or "").split("<BR>"))
+        lines.extend((p.get("infrastruktur") or "").split("<BR>"))
+        lines = [l.strip() for l in lines]
+        return " / ".join([l for l in lines if l])
+
+    def parse_opening_hours(self, p):
+        if "ist rund um die Uhr besetzt" in (p.get("bemerkung") or ""):
+            return "24/7"
+        oh = OpeningHours()
+        for day in "Mo Di Mi Do Fr Sa So".split():
+            # The counter (Schalter) can have shorter opening hours
+            # than the building (Gebäude) that contains the counter.
+            # We return the hours of the counter, but they aren’t set,
+            # let’s fall back to the opening hours of the building.
+            for part in ("schalter", "gebaeude"):  # counter, building
+                if h := p.get(f"oeffnungszeiten_{part}_{day.lower()}"):
+                    if h.count(";") == 3:
+                        open_am, close_am, open_pm, close_pm = h.split(";")
+                        if close_am and open_pm:
+                            oh.add_range(
+                                DAYS_DE[day], open_am, close_am, time_format="%H.%M"
+                            )
+                            oh.add_range(
+                                DAYS_DE[day], open_pm, close_pm, time_format="%H.%M"
+                            )
+                        else:
+                            oh.add_range(
+                                DAYS_DE[day], open_am, close_pm, time_format="%H.%M"
+                            )
+                        break
+        return oh.as_opening_hours()
+
+    def parse_park(self, p):
+        return {"leisure": "park"}
+
+    def parse_phone(self, p):
+        phone = (p.get("tel") or "").replace(" ", "")
+        if phone.startswith("0"):
+            phone = "+41" + phone[1:]
+        return phone
```

Comment:
This is great and diligent. I've always kind of resisted doing the phone "clean" as I feel a more structural pipeline approach will do it for us. This needs knowing the country code. But hey, you're already guiding towards a happier place in this context with:
https://github.com/alltheplaces/alltheplaces/pull/4229

Reply:
Once we know the country code (after #4229 is merged), normalizing phone numbers would actually be quite easy. I’ve added a TODO comment.